app/models/branch_of_service.rb

```diff
@@ -6,6 +6,8 @@ class BranchOfService < ApplicationRecord
   has_one :branch_of_service_contact, dependent: :destroy
   has_many :service_specific_posts, dependent: :destroy
 
+  accepts_nested_attributes_for :branch_of_service_contact, :service_specific_posts
```

Comment:
[The `accepts_nested_attributes_for` class method](http://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html) allows a parent class to include data for a relation (or relations) during the create process (as `*_attributes`):

```rb
BranchOfService.create(name: 'Army', display_order: 1, branch_of_service_contact_attributes: {
  custsvc_org 'My Army organization',
  # etc. etc. etc.
})
```